docs/conf.py

```diff
@@ -0,0 +1,56 @@
+"""Sphinx documentation generator configuration file.
+
+The full set of configuration options is listed on the Sphinx website:
+http://sphinx-doc.org/config.html
+
+"""
+import sys
+import os
+
+
+# Add the AirGun root directory to the system path. This allows references
+# such as :mod:`airgun.browser` to be processed correctly.
+sys.path.insert(
+    0,
+    os.path.abspath(os.path.join(
+        os.path.dirname(__file__),
+        os.path.pardir
+    ))
+)
+
+# Project Information ---------------------------------------------------------
+
+project = 'AirGun'
+copyright = '2018, Andriy Balakhtar'
+version = '0.0.1'
+release = version
+
+# General Configuration -------------------------------------------------------
+
+extensions = [
+    'sphinx.ext.autodoc',
+]
+source_suffix = '.rst'
+master_doc = 'index'
+exclude_patterns = ['_build']
+nitpicky = True
+nitpick_ignore = [
+    ('py:class', 'widgetastic.browser.Browser'),
```

Comment:
Take a look at `sphinx.ext.intersphinx`, maybe it will allow us to use proper references. If not - i'd rather use 
```
``widgetastic.browser.Browser``
```
instead of 
```
:class:`widgetastic.browser.Browser`
```
not to have to ignore such references manually. At least that's how we handle such references in other projects.

Reply:
OK, I have spent a lot of time with this and using `sphinx.ext.intersphinx` I was able to get rid of these ignores:

```
    ('py:class', 'bool'),
    ('py:class', 'dict'),
    ('py:class', 'str'),
```

but I was unable to resolve:

 * unable to resolve class internal references like `('py:meth', 'current_org')`. Found https://stackoverflow.com/questions/21289806/link-to-class-method-in-python-docstring but that did not helped me.
 * unable to resolve references to `widgetastic` and `navmazing` (and two cases of `airgun`, but respective class inherits from `widgetastic` and `navmazing`). I believe it is because of https://github.com/RedHatQE/widgetastic.core/issues/86 (missing `objects.inv` file there) and https://github.com/RedHatQE/navmazing/issues/18.

So possibilities are:

 * keep these in the whitelist
 * remove from whitelist and change syntax in the docstring as you mentioned

I'll probably create two new issues, one to track class internal references and second once docs is fixed on `widgetastic` and `navmazing` issues.

Reply:
Right, you're trying to handle way too many issues in a single PR. Just submit a new issue not to forget about that and to track the progress. For "Init" PR current progress is more than enough ;)